packages/@aws-cdk/aws-cognito/lib/user-pool-idps/oidc.ts

```diff
@@ -0,0 +1,130 @@
+import { Construct } from 'constructs';
+import { CfnUserPoolIdentityProvider } from '../cognito.generated';
+import { UserPoolIdentityProviderProps } from './base';
+import { UserPoolIdentityProviderBase } from './private/user-pool-idp-base';
+
+/**
+ * Properties to initialize UserPoolIdentityProviderOidc
+ */
+export interface UserPoolIdentityProviderOidcProps extends UserPoolIdentityProviderProps {
+  /**
+   * The name of the provider
+   */
+  readonly name: string;
+
+  /**
+   * The client id
+   */
+  readonly clientId: string;
+
+  /**
+   * The client secret
+   */
+  readonly clientSecret: string;
+
+  /**
+   * The OAuth 2.0 scopes that you will request from OpenID Connect. Scopes are
+   * groups of OpenID Connect user attributes to exchange with your app.
+   *
+   * @default ['openid']
+   */
+  readonly scopes?: string[];
+
+  /**
+   * Identifiers
+   *
+   * Identifiers can be used to redirect users to the correct IdP in multitenant apps.
+   *
+   * @default - no identifiers used
+   */
+  readonly identifiers?: string[]
+
+  /**
+   * The method to use to request attributes
+   *
+   * @default OidcAttributeRequestMethod.GET
+   */
+  readonly attributeRequestMethod?: OidcAttributeRequestMethod
+
+  /**
+   * Issuer URL
+   */
+  readonly issuerUrl: string;
+
+  /**
+   * OpenID connect endpoints
+   *
+   * @default - auto discovered with issuer URL
+   */
+  readonly endpoints?: OidcEndpoints;
+}
+
+/**
+ * OpenID Connect endpoints
+ */
+export interface OidcEndpoints {
+  /**
+   * Authorization endpoint
+   */
+  readonly authorization: string;
+
+  /**
+    * Token endpoint
+    */
+  readonly token: string;
+
+  /**
+    * UserInfo endpoint
+    */
+  readonly userInfo: string;
+
+  /**
+    * Jwks_uri endpoint
+   */
+  readonly jwksUri: string;
+}
+
+/**
+ * The method to use to request attributes
+ */
+export enum OidcAttributeRequestMethod {
+  /** GET */
+  GET = 'GET',
+  /** POST */
+  POST = 'POST'
+}
+
+/**
+ * Represents a identity provider that integrates with OpenID Connect
+ * @resource AWS::Cognito::UserPoolIdentityProvider
+ */
+export class UserPoolIdentityProviderOidc extends UserPoolIdentityProviderBase {
+  public readonly providerName: string;
+
+  constructor(scope: Construct, id: string, props: UserPoolIdentityProviderOidcProps) {
+    super(scope, id, props);
+
+    const scopes = props.scopes ?? ['openid'];
+
+    const resource = new CfnUserPoolIdentityProvider(this, 'Resource', {
+      userPoolId: props.userPool.userPoolId,
+      providerName: props.name,
```

Comment:
Would it be better to use `physicalName` here and generate one if the user doesn't provide one? It doesn't look like OIDC has a specific name requirement like the others.

Reply:
ahh I missed that it wasn't extending `Resource` directly. How about:

```ts
providerName: props.name || Lazy.string({ produce: () => Names.uniqueId(this) }),
```

Reply:
`id` might be a better default since the provider name is limited to 32 chars?